src/core/openthread.cpp

```diff
@@ -1256,9 +1256,17 @@ void otSetReceiveIp6DatagramFilterEnabled(otInstance *aInstance, bool aEnabled)
 ThreadError otSendIp6Datagram(otInstance *aInstance, otMessage aMessage)
 {
     ThreadError error;
+    Message *message;
+    Ip6::Header header;
 
     otLogFuncEntry();
 
+    message = static_cast<Message *>(aMessage);
+    assert(message->GetLength() >= sizeof(header));
+
+    message->Read(0, sizeof(header), &header);
+    assert(!header.GetDestination().IsRealmLocalMulticast() || header.GetNextHeader() != Ip6::kProtoHopOpts);
```

Comment:
Shouldn't we remove this assert now? On the receive side, do we remove the MPL Option header, so that between Source and Destination, the packet looks unchanged?